megaavr/cores/dxcore/wiring_analog.c

```diff
@@ -888,6 +887,39 @@ void analogReadEnableDifferential(uint8_t negPin) {
   ADC0.MUXNEG = negPin;
 }
 
+// Designed expecting user to pass in number of results, not number set to the register
+void analogReadSampleNum(uint8_t numSamples) {
+  while (ADC0.COMMAND & ADC_STCONV_bm) {}
+    ADC0.CTRLB &= 0b11111000;   //Setting accumulation to 0
+    switch (numSamples) {
+    case 2:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC2_gc;
+      break;
+    case 4:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC4_gc;
+      break;
+    case 8:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC8_gc;
+      break;
+    case 16:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC16_gc;
+      break;
+    case 32:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC32_gc;
+      break;
+    case 64:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC64_gc;
+      break;
+    case 128:
+      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC128_gc;
+      break;
+    default:
+        badArg("Unexpected value");
```

Comment:
One last thing, I think you'll need a check for 0 since otherwise this error will be given if `numSamples` is 0. You could just add an empty 0 case.